Basic.Azure.Storage/Extensions/BlobServiceClientExtensions.cs

```diff
@@ -167,6 +163,15 @@ private static string GenerateBlockId()
             return await Task.Run(() => Convert.ToBase64String(MD5.Create().ComputeHash(fullData, offset, length)));
         }
 
+        private static T RunSynchronously<T>(Func<Task<T>> function)
+        {
+            var smartTask = Task.Run(function);
+
+            // GetAwaiter().GetResult() doesn't wrap the underlying exception in an AggregateException
+            // http://blogs.msdn.com/b/pfxteam/archive/2011/09/28/task-exception-handling-in-net-4-5.aspx
+            return smartTask.GetAwaiter().GetResult();
```

Comment:
This uses the new await behavior (not wrapping in Aggregates when unnecessary) instead of the old .Result behavior.